### PR TITLE
[3.4] bpo-26617: Ensure gc tracking is off when invoking weakref callbacks.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
(cherry picked from commit 8f657c35b978b681e6e919f08358992e1aed7dc1)

### CODE REVIEW

File: Misc/NEWS [hunk 1]
```diff
@@ -10,6 +10,8 @@ Release date: XXXX-XX-XX
 Core and Builtins
 -----------------
 
+- Issue #26617: Fix crash when GC runs during weakref callbacks.
```
Comment (vstinner) [inline Misc/NEWS]: You may rewrite this as a NEWS.d entry with blurb, to prevent conflicts.

**Review by vstinner — APPROVED** [review]

**Comment by vstinner** [comment]
You might wait for my PR #2475 to be merged to get a CI on 3.4 ;-)

File: Misc/NEWS (hunk 1 above)
Comment (larryhastings) [reply Misc/NEWS]: Yes, please do this.

Comment (serhiy-storchaka) [reply Misc/NEWS]: I tried to use blurb, entered all data, but it doesn't have any effect.

Comment (larryhastings) [reply Misc/NEWS]: If you ran blurb and entered data, and it didn't print an error, then it had an effect.  It creates a file in Misc/NEWS.d/next and runs "git add" on that file.

Were you inside this repo when you ran it?

If it's more convenient, you could copy an existing file in Misc/NEWS.d/next and manually change the filename to match the current time.  But we should figure out why blurb isn't working for you.

Comment (serhiy-storchaka) [reply Misc/NEWS]: Ah, this is my fault! Sorry. I executed `git diff` and `git log`, but their didn't show any changes. I needed to use `git diff --staged` or `git status` instead.

**Comment by larryhastings** [comment]
Thanks for the backport!

**Pull request merged**